software/25.Flutter/README.md

```diff
@@ -0,0 +1,195 @@
+
+Lets take a look of how to do list with Flutter. For it you have muliple choices, here are few of them:
+
+- [List.generate](https://api.flutter.dev/flutter/dart-core/List/List.generate.html)
+
+
+    As there is no exemple with `widget` in the doc, here is one: 
+    
+    ```dart
+    Column(
+        children: List.generate(
+            10,
+            (index) {
+                return Text(index.toString());
+            }
+        ),
+    )
+    ```
+- [ListView.builder](https://docs.flutter.dev/cookbook/lists/long-lists)
+
+- [For loop](https://stackoverflow.com/questions/56947046/flutter-for-loop-to-generate-list-of-widgets)
+
+You have to make sure you component is workink by calling your component with this list in parameters:
+
+```dart
+final List<String> _arr = ["Todo 1", "Todo 2", "Todo 3", "Todo 4", "Todo 5"];
+```
+
+:warning: If the list can't be display to the entire screen (it will happen), you might wrap your list with [SingleChildScrollView](https://api.flutter.dev/flutter/widgets/SingleChildScrollView-class.html), to make your screen scrollable.
+
+## Step 3 - Models
+
+For this step you have to create a folder `lib/models` and a `todo.dart`.
+
+The goal of this step is to create a class that will define the atributs of a todo and impelement it.
+
+Your todo class need to contain the following attributs:
+  
+  - A required string `title`
+  - An optional string `description`
+
+Then you have to replace the list of string `_arr` that you passed as parameters of your `ToDoList component` by:
+
+```dart
+final List<Todo> _todos = [Todo(title: "Todo 1"), Todo(title: "Todo 2", description: "description of todo 2"), Todo(title: "Todo 3"), Todo(title: "Todo 4"), Todo(title: "Todo 5", description: "description of todo 5")];
+```
+
+Then your `ToDoList component` have now to display the title of all todos.
+
+## Step 4 - Card and ListTile
+
+In order to make the work easier, you will use a [list tile](https://api.flutter.dev/flutter/material/ListTile-class.html). It will help you to organise your todo.
+
+So you can set the title of your todo as the title of your `tile` and the `description` of the todo as the subtitle (but you are free to whatever you want).
+
+That seem a little bit better but we can do more. Make your list more eye-catching by wraping each element of your list with a [card](https://api.flutter.dev/flutter/material/Card-class.html).
+
+As now you have a `list tile` you can already make the delete [icon button](https://api.flutter.dev/flutter/material/IconButton-class.html) in the trailing of your tile.
+
+:warning: In flutter you have to [refresh the state of the page](https://api.flutter.dev/flutter/widgets/State/setState.html) after modifing a displayed variable to see it.
+
+## Step 5 - Create a todo
+
+At this time, you might have the display and the delete of your todos, the only thing that miss is the todo creation page.
```

Comment:
Either "the display and the deletion" or "display and deletion function/features "